tools/gfx/vulkan/vk-shader-program.cpp

```diff
@@ -74,9 +74,10 @@ Result ShaderProgramImpl::createShaderModule(
     // to control the entry point naming behavior in SPIRV-direct path
     // so we can remove the ad-hoc logic here.
     auto realEntryPointName = entryPointInfo->getNameOverride();
-    const char* spirvBinaryEntryPointName = "main";
-    if (m_device->m_desc.slang.targetFlags & SLANG_TARGET_FLAG_GENERATE_SPIRV_DIRECTLY)
-        spirvBinaryEntryPointName = realEntryPointName;
+    const char* spirvBinaryEntryPointName =
```

Comment:
Our spirv backend should respect the new `-fvk-use-entrypoint-name` option and use "main" if the option is not set.